extensions/pkg/controller/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,337 @@
+				Name: namespaceName,
+			},
+		}
+
+		cluster = &extensionsv1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+			Spec: extensionsv1alpha1.ClusterSpec{
+				CloudProfile: runtime.RawExtension{Raw: []byte("{}")},
+				Seed:         runtime.RawExtension{Raw: []byte("{}")},
+				Shoot:        runtime.RawExtension{Raw: []byte("{}")},
+			},
+		}
+
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      v1beta1constants.SecretNameCloudProvider,
+				Namespace: namespaceName,
+			},
+		}
+
+		backupBucket = &extensionsv1alpha1.BackupBucket{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+			Spec: extensionsv1alpha1.BackupBucketSpec{
+				DefaultSpec: extensionsv1alpha1.DefaultSpec{
+					Type: test.Type,
+				},
+				SecretRef: corev1.SecretReference{
+					Name:      v1beta1constants.SecretNameCloudProvider,
+					Namespace: namespaceName,
+				},
+				Region: "foo",
+			},
+		}
+		backupBucketObjectKey = client.ObjectKey{Name: namespaceName}
+	)
+
+	var cleanupHandle framework.CleanupActionHandle
+	cleanupHandle = framework.AddCleanupAction(func() {
+		By("delete backupbucket")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, backupBucket))).To(Succeed())
+
+		By("wait until backupbucket is deleted")
+		Expect(common.WaitUntilExtensionCRDeleted(
+			ctx,
+			c,
+			logger,
+			func() extensionsv1alpha1.Object { return &extensionsv1alpha1.BackupBucket{} },
+			extensionsv1alpha1.BackupBucketResource,
+			backupBucket.Namespace,
+			backupBucket.Name,
+			time.Second,
+			3*time.Second,
+		)).NotTo(HaveOccurred())
+
+		Expect(c.Get(ctx, client.ObjectKey{Name: namespaceName}, namespace)).To(Succeed())
+		Expect(namespace.Annotations[test.AnnotationKeyOperation]).To(Equal(test.AnnotationValueOperationDelete))
+
+		By("delete secret")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, secret))).To(Succeed())
+
+		By("delete cluster")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+
+		By("delete namespace")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+
+		framework.RemoveCleanupAction(cleanupHandle)
+	})
+
+	By("create namespace for test execution")
+	Expect(c.Create(ctx, namespace)).To(Succeed())
+
+	By("create cluster")
+	Expect(c.Create(ctx, cluster)).To(Succeed())
+
+	By("create cloudprovider secret into namespace")
+	Expect(c.Create(ctx, secret)).To(Succeed())
+
+	By("create backupbucket")
+	timeIn1 := time.Now().String()
+	metav1.SetMetaDataAnnotation(&backupBucket.ObjectMeta, test.AnnotationKeyTimeIn, timeIn1)
+	Expect(c.Create(ctx, backupBucket)).To(Succeed())
+
+	By("wait until backupbucket is ready")
+	Expect(waitForBackupBucketToBeReady(ctx, c, logger, backupBucket)).To(Succeed())
+
+	By("verify secret handling")
+	Expect(c.Get(ctx, client.ObjectKey{Namespace: secret.Namespace, Name: secret.Name}, secret)).To(Succeed())
+	Expect(secret.Finalizers).To(ConsistOf(backupbucketcontroller.FinalizerName))
```

Comment:
Can you avoid using constants from the production code throughout the test?
```suggestion
	Expect(secret.Finalizers).To(ConsistOf("extensions.gardener.cloud/backupbucket"))
```